hoomd/SnapshotSystemData.cc

```diff
@@ -39,6 +39,16 @@ void SnapshotSystemData<Real>::replicate(unsigned int nx, unsigned int ny, unsig
     pair_data.replicate(n, old_n);
     }
 
+template<class Real> void SnapshotSystemData<Real>::wrap()
+    {
+    for (unsigned int i = 0; i < particle_data.size; i++)
+        {
+        vec3<double> p = particle_data.pos[i];
+        global_box.wrap(p, particle_data.image[i]);
```

Comment:
```suggestion
        global_box.wrap(p, make_int3(0, 0, 0));
```
I'm not sure if wrapping the particles and counting the image can lead to problems. It really depends when the wrap feature is applied. If it is applied to make sure that this snapshot can be used as an initial configuration, I reckon that updating `particle_data.image[i]` messes up mean squared displacement calculations, for example. Or are the images set to zero before the simulation starts? 
For a snapshot after a simultion it makes sense to update the images. But here this feature is probably unneccesary, as HOOMD wraps the particles in the simulations and no particle should be outside the simulation box anyway.

Reply:
We do need to update the image to preserve the initial unwrapped position of the particle as the user provided it to us. For example, say a user creates molecules where the center of the molecule is in the box but part of the molecule leaves the box. Then they call wrap. After calling wrap, the molecule is broken across the box boundary, but the image flags can be used to unwrap the particles so the molecule is contiguous. 

This is the use case in https://github.com/mosdef-hub/mbuild/pull/871 where the new `snapshot.wrap` method will be used.

If a user desires all images to be reset to 0, then they can do so after calling wrap. We could add a note in the documentation for `wrap` that it modifies both `position` and `image`.